File: CRM/backend/src/models/Company.js
import mongoose from 'mongoose';

const companySchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
      trim: true,
      unique: true,
    },
    domain: {
      type: String,
      trim: true,
    },
    adminId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    isActive: {
      type: Boolean,
      default: true,
    },
  },
  { timestamps: true }
);

companySchema.index({ adminId: 1 });

export const Company = mongoose.model('Company', companySchema);

File: CRM/Client-app/app/src/main/java/com/example/crmprime/data/model/Company.kt
package com.example.crmprime.data.model

import com.google.gson.annotations.SerializedName

data class Company(
    @SerializedName("_id")
    val id: String,
    @SerializedName("name")
    val name: String,
    @SerializedName("domain")
    val domain: String? = null,
    @SerializedName("adminId")
    val adminId: String,
    @SerializedName("isActive")
    val isActive: Boolean = true
)

